src/renderer/views/swap/SwapView.tsx

```diff
@@ -168,6 +168,20 @@ export const SwapView: React.FC<Props> = (_): JSX.Element => {
     reloadInboundAddresses()
   }, [reloadBalances, reloadInboundAddresses, reloadPools])
 
+  const getStoredSlipTolerance = (): SlipTolerance =>
+    FP.pipe(
+      localStorage.getItem(SLIP_TOLERANCE_KEY) as string,
+      O.fromNullable,
+      O.map((s) => {
+        const slipTolerance = +s as SlipTolerance
+        changeSlipTolerance(slipTolerance)
+        return slipTolerance
```

Comment:
How about the following to have stronger types? 

1. Define a type guard `isSlipTolerance` in `src/shared/api/types.ts`

```ts
export const isSlipTolerance = (value: SlipTolerance | number): value is SlipTolerance => {
  const t: SlipTolerance[] = [3, 5, 10]
  return t.includes(value as SlipTolerance)
}
``` 

2. to use it here

```ts
      O.map((s) => {
        const itemAsInt = parseInt(s)
        const slipTolerance = isSlipTolerance(itemAsInt) ? itemAsInt : DEFAULT_SLIP_TOLERANCE
        changeSlipTolerance(slipTolerance)
        return slipTolerance
      }),
```

Reply:
BTW: Pls move `SlipTolerance` + `isSlipTolerance` from (1) `shared/api/types` to (2) `src/renderer/types/asgardex.ts`. We use (1) for sharing code in `main` and `renderer` processes in Electron, but those types are needed for `renderer` only

Reply:
Makes sense changes are on the way